the-blue-alliance-ios/View Elements/Match/Breakdown/MatchBreakdownConfigurator2020.swift

```diff
@@ -0,0 +1,258 @@
+        rows.append(row(title: "Ranking Points", key: "rp", formatString: "+%@ RP", red: red, blue: blue))
+
+        // Clean up any empty rows
+        let validRows = rows.compactMap({ $0 })
+        if !validRows.isEmpty {
+            snapshot.appendSections([nil])
+            snapshot.appendItems(validRows)
+        }
+    }
+
+    private static func initLine(red: [String: Any]?, blue: [String: Any]?) -> BreakdownRow? {
+        var redLineStrings: [String] = []
+        var blueLineStrings: [String] = []
+        
+        for i in [1, 2, 3] {
+            guard let initValues = values(key: "initLineRobot\(i)", red: red, blue: blue) else {
+                return nil
+            }
+            let (rv, bv) = initValues
+            guard let redInit = rv as? String, let blueInit = bv as? String else {
+                return nil
+            }
+            redLineStrings.append(redInit)
+            blueLineStrings.append(blueInit)
+        }
+        
+        let elements = [redLineStrings, blueLineStrings].map { (lineStrings) -> [AnyHashable] in
+            return lineStrings.map { (line) -> AnyHashable in
+                switch line {
+                case "None":
+                    return BreakdownStyle.xImage
+                case "Exited":
+                    return BreakdownStyle.checkImage
+                default:
+                    return "?"
+                }
+            }
+        }
+        
+        var (redElements, blueElements) = (elements[0], elements[1])
+
+        // Add the point totals for the init line
+        guard let initLinePoints = values(key: "autoInitLinePoints", red: red, blue: blue) else {
+            return nil
+        }
+        
+        let (redLinePoints, blueLinePoints) = initLinePoints
+        redElements.append("(+\(redLinePoints ?? 0))")
+        blueElements.append("(+\(blueLinePoints ?? 0))")
+
+        return BreakdownRow(title: "Initiation Line exited", red: redElements, blue: blueElements)
+    }
+
+    private static func powerCellRow(title: String, period: String, red: [String: Any]?, blue: [String: Any]?) -> BreakdownRow? {
+        let keys = ["CellsBottom", "CellsOuter", "CellsInner"]
+        let images = [BreakdownStyle2020.bottomImage, BreakdownStyle2020.outerImage, BreakdownStyle2020.innerImage]
+        let locations = keys.map { "\(period)\($0)" }
+        
+        var redCells: [Int] = []
+        var blueCells: [Int] = []
+        
+        for location in locations {
+            guard let cellValues = values(key: location, red: red, blue: blue) else {
+                return nil
+            }
+
+            let (rv, bv) = cellValues
+
+            guard let redCellValue = rv as? Int, let blueCellValue = bv as? Int else {
+                return nil
+            }
+            redCells.append(redCellValue)
+            blueCells.append(blueCellValue)
+        }
+        
+        let redValues = zip((images).map {
+            return makeImageView(image: $0)
```

Comment:
We actually need to set the `contentMode` on this one to either be aspect fit or aspect fill. These are getting stretched. It's SUPER subtle - but they are. In the screenshots below, the first one is the existing image (default mode is scale to fill), and the second one is with aspect fit.

<img width="513" alt="Screen Shot 2020-02-16 at 7 36 17 PM" src="https://user-images.githubusercontent.com/516458/74616089-d93ab600-50f3-11ea-97d9-2cc358ba261e.png">
<img width="494" alt="Screen Shot 2020-02-16 at 7 36 24 PM" src="https://user-images.githubusercontent.com/516458/74616093-db047980-50f3-11ea-86bb-d937009b6c81.png">

Reply:
To see if need aspect fill, test on an iPad Sim and see if things look okay if we have all 0s. If it looks weird or the images are WAY too big because they're being too greedy with their frame, we might need aspect fit or center or something.

Reply:
Actually here we go

## Aspect Fill
![Simulator Screen Shot - iPad Pro (12 9-inch) (3rd generation) - 2020-02-16 at 19 42 32](https://user-images.githubusercontent.com/516458/74616219-91685e80-50f4-11ea-9208-bc63fe22847b.png)
![Simulator Screen Shot - iPhone 11 Pro - 2020-02-16 at 19 43 31](https://user-images.githubusercontent.com/516458/74616237-a34a0180-50f4-11ea-86ce-f905441e64ff.png)

## Aspect Fit
![Simulator Screen Shot - iPad Pro (12 9-inch) (3rd generation) - 2020-02-16 at 19 46 15](https://user-images.githubusercontent.com/516458/74616309-09368900-50f5-11ea-8500-8c35d07db493.png)
![Simulator Screen Shot - iPhone 11 Pro - 2020-02-16 at 19 44 35](https://user-images.githubusercontent.com/516458/74616281-dd1b0800-50f4-11ea-8fc5-f3dce8b0b30c.png)

Notice the images look pretty large even in tight situations using aspect fill, but the left-to-right spacing is a little weird. I think it's worth though to make these images a little bit more visible while maintaining the proper aspect ratio for the shapes.